include/emqx.hrl

```diff
@@ -145,8 +145,10 @@
 
 -record(banned, {
           who    :: {clientid,  binary()}
+                  | {peerhost, inet:ip_address()}
                   | {username,   binary()}
-                  | {ip_address, inet:ip_address()},
+                  | {ip_address, inet:ip_address()}
+                  | undefined,
```

Comment:
allowing `undefined` to be banned does not seem reasonable to me.

Reply:
who is not always defined causing record construction error in dialyzer